walnut/converter/converter.go

```diff
@@ -0,0 +1,156 @@
+}
+
+// return pointer to string
+func createString(x string) *string {
+	return &x
+}
+
+// ExtractTaskClasses takes in a DPL Dump string and extracts
+// an array of Tasks
+func ExtractTaskClasses(DPL Dump) (tasks []*task.Class, err error) {
+
+	for index := range DPL.Workflows {
+		var channelName string
+		if index+1 == len(DPL.Workflows) {
+			channelName = DPL.Workflows[index].Name
+		} else {
+			channelName = "from_" + DPL.Workflows[index].Name + "_to_" + DPL.Workflows[index+1].Name
+		}
+
+		taskName := DPL.Workflows[index].Name
+		defaultBindChannel := channel.Inbound{
+			Channel: channel.Channel{
+				Name:        channelName,
+				Type:        channel.ChannelType("push"), // defaulting to push
+				SndBufSize:  1000,
+				RcvBufSize:  1000,
+				RateLogging: 60,
+				Transport:   channel.TransportType("shmem"),
+			},
+			Addressing: "ipc",
+		}
+
+		// Not required for Task Templates
+		/*
+			defaultConnectChannel := channel.Outbound{
+				Channel: channel.Channel{
+					Name:        workflowName,
+					Type:        channel.ChannelType(""),
+					SndBufSize:  1000,
+					RcvBufSize:  1000,
+					RateLogging: 60,
+					Transport:   channel.TransportType("shmem"),
+				},
+				// Target: "", No default value
+			}
+		*/
+
+		task := task.Class{
+			Identifier: task.TaskClassIdentifier{
+				Name: taskName,
+			},
+			Defaults: gera.MakeStringMapWithMap(map[string]string{
+				"user": "flp",
+			}),
+			Control: struct {
+				Mode controlmode.ControlMode "yaml:\"mode\""
+			}{Mode: controlmode.FAIRMQ},
+			Command: &common.CommandInfo{
+				// Env: []string, -> Default to empty array
+				Shell:     createBool(true),
+				Value:     &DPL.Metadatas[index+1].Executable,
+				Arguments: DPL.Metadatas[index+1].CmdlLineArgs,
+				User:      createString("flp"),
+			},
+			Wants: task.ResourceWants{
+				Cpu:    createFloat(0.15),
+				Memory: createFloat(128),
+				Ports:  task.Ranges{}, //begin - end OR range
+			},
+			Bind: []channel.Inbound{defaultBindChannel},
+			Properties: gera.MakeStringMapWithMap(map[string]string{
+				"severity": "trace",
+				"color":    "false",
+			}),
+			// Connect:   []channel.Outbound{defaultConnectChannel},
+		}
+		// fmt.Printf("\nTASK:\n%v\n", task)
+		tasks = append(tasks, &task)
+	}
+	return tasks, nil
+}
+
+// TaskToYAML takes as input an array of pointers to task.Class
+// and writes them to a AliECS friendly YAML file
+func TaskToYAML(extractedTasks []*task.Class) (err error) {
+
+	// Check if "tasks" directory exists. If not, create it
+	_, err = os.Stat("tasks")
+	if os.IsNotExist(err) {
+		errDir := os.MkdirAll("tasks", 0755)
+		if errDir != nil {
+			return fmt.Errorf("create dir failed: %w", err)
+		}
+	}
+
+	for _, SingleTask := range extractedTasks {
+		YAMLData, err := yaml.Marshal(SingleTask)
+		if err != nil {
+			return fmt.Errorf("marshal failed: %w", err)
+		}
+
+		// Write marshaled YAML to file
+		err = ioutil.WriteFile("tasks/"+SingleTask.Identifier.Name+".yaml", YAMLData, 0755)
```

Comment:
`0755` is appropriate for a directory but not for a file, which should default to `0644`. The reason for this is that the `x` permission means execute for files, but `cd` for directories.